### ISSUE TITLE
inspec containers newer then 4.18.51 exhibit issues when passing --sudo flag

### ISSUE THREAD

**Issue by n0coast** [inspec/inspec#4899]
The docker container chef/inspec changed behavior after 4.18.51 when run against a remote ssh target and passing the --sudo flag

# Describe the problem
The docker container chef/inspec changed behavior **after** 4.18.51.
Starting in 4.18.52 when running  exec against a remote ssh target and passing the --sudo flag the run will hang at `cmd = sh -c '(sudo -v) < /dev/null'`.
Starting in 4.18.65 the run will continue past the `sudo -v` step but will error out complaining that sudo requires a tty: `sudo: no tty present and no askpass program specified`

This behavior is experienced against AmazonLinux2 EC2 targets, where the docker command was run from either OSX or AmazonLinux2.

The following 3 runs exhibit the differing behaviors at the highest version that bahavior was found at if that makes sense. 

All runs were against the exact same target host with no configuration changes to the target between the runs.

Here are the relatively simple tests being run in az2_base_test.rb:
```
describe service('filebeat') do
  it { should be_enabled }
  it { should be_installed }
  it { should be_running }
end

describe command('getenforce') do
  its('exit_status') {should eq 0}
  its('stdout') { should eq "Enforcing\n" }
end

describe iptables do
  it { should have_rule('-P INPUT DROP') }
end
```

Here is a run using 4.18.51 (the last version where this worked):
```
docker run --rm -v $(pwd)/../inspec_files/:/share:Z chef/inspec:4.18.51 exec -i secret_secret --sudo --target=ssh://ec2-user@10... --chef-license=accept-no-persist --no-enable-telemetry -l debug --diagnose az2-base-test.rb
InSpec version: 4.18.51
Train version: 3.2.0
Command line configuration:
{"enable_telemetry"=>false,
 "winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "key_files"=>["secret_secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@10...",
 "chef_license"=>"accept-no-persist",
 "log_level"=>"debug",
 "diagnose"=>true}
JSON configuration file:
{"version"=>"1.2"}
Merged configuration:
{"reporter"=>["cli"],
 "show_progress"=>false,
 "color"=>true,
 "create_lockfile"=>true,
 "backend_cache"=>true,
 "enable_telemetry"=>false,
 "winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "key_files"=>["secret_secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@10...",
 "chef_license"=>"accept-no-persist",
 "log_level"=>"debug",
 "diagnose"=>true}

D, [2020-02-07T22:13:14.655153 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (cmd.exe /c ver)
D, [2020-02-07T22:13:14.655314 #1] DEBUG -- : [SSH] opening connection to ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}>
D, [2020-02-07T22:13:15.003221 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (Get-WmiObject Win32_OperatingSystem | Select Caption,Version | ConvertTo-Json)
D, [2020-02-07T22:13:15.050418 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (uname -s)
D, [2020-02-07T22:13:15.094766 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (uname -m)
D, [2020-02-07T22:13:15.139123 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /etc/debian_version && cat /etc/debian_version)
D, [2020-02-07T22:13:15.183793 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /etc/os-release && cat /etc/os-release)
D, [2020-02-07T22:13:15.230657 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /usr/bin/FastCli)
D, [2020-02-07T22:13:15.274714 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (show version | json)
D, [2020-02-07T22:13:15.320047 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /etc/lsb-release && cat /etc/lsb-release)
D, [2020-02-07T22:13:15.365900 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /usr/bin/lsb-release && cat /usr/bin/lsb-release)
D, [2020-02-07T22:13:15.410399 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /etc/oracle-release && cat /etc/oracle-release)
D, [2020-02-07T22:13:15.454716 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /etc/enterprise-release && cat /etc/enterprise-release)
D, [2020-02-07T22:13:15.500106 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /etc/parallels-release && cat /etc/parallels-release)
D, [2020-02-07T22:13:15.545213 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (test -f /etc/system-release && cat /etc/system-release)
D, [2020-02-07T22:13:15.591466 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (sudo echo)
[2020-02-07T22:13:15+00:00] DEBUG: Option backend_cache is enabled
[2020-02-07T22:13:15+00:00] DEBUG: Resolve az2-base-test.rb into cache /root/.inspec/cache
[2020-02-07T22:13:15+00:00] DEBUG: Dependency does not exist in the cache az2-base-test.rb
[2020-02-07T22:13:15+00:00] DEBUG: Starting run with targets: ["Inspec::Profile<tests from az2-base-test.rb>"]
[2020-02-07T22:13:15+00:00] DEBUG: Loading az2-base-test.rb into #<Inspec::ProfileContext:0x000055e4e26ed418>
[2020-02-07T22:13:15+00:00] DEBUG: Registering rule (generated from az2-base-test.rb:1 0408fd36e5dcae04ae3d600996320794)
[2020-02-07T22:13:15+00:00] DEBUG: Registering rule (generated from az2-base-test.rb:7 890b9b00230715259635cab63381a9cb)
[2020-02-07T22:13:15+00:00] DEBUG: Registering rule (generated from az2-base-test.rb:12 abb0aaa2db92d24ec4743e00d3a1d8a6)
D, [2020-02-07T22:13:15.740887 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (systemctl show --no-pager --all filebeat)
D, [2020-02-07T22:13:15.811260 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (systemctl is-active filebeat --quiet)
D, [2020-02-07T22:13:15.876327 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (systemctl is-enabled filebeat --quiet)
D, [2020-02-07T22:13:15.965128 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (getenforce)
D, [2020-02-07T22:13:16.024278 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (bash -c 'type "/usr/sbin/iptables"')
D, [2020-02-07T22:13:16.083622 #1] DEBUG -- : [SSH] ec2-user@10...<{:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}> (/usr/sbin/iptables  -S)

Profile: tests from az2-base-test.rb (tests from az2-base-test.rb)
Version: (not specified)
Target:  ssh://ec2-user@10...:22

  Service filebeat
     ✔  is expected to be enabled
     ✔  is expected to be installed
     ✔  is expected to be running
  Command: `getenforce`
     ✔  exit_status is expected to eq 0
     ✔  stdout is expected to eq "Enforcing\n"
  Iptables
     ✔  is expected to have rule "-P INPUT DROP"
```

Here is a run using 4.18.61 (The last version where inspec would just "hang"), note that these runs would need to be stopped with ctrl-c:
```
docker run --rm -v $(pwd)/../inspec_files/:/share:Z chef/inspec:4.18.61 exec -i secret_secret --sudo --target=ssh://ec2-user@10... --chef-license=accept-no-persist --no-enable-telemetry -l debug --diagnose az2-base-test.rb
Unable to find image 'chef/inspec:4.18.61' locally
4.18.61: Pulling from chef/inspec
e6b0cf9c0882: Already exists
39db6355b3f6: Already exists
e92d591b4e43: Already exists
f89afb0a171d: Already exists
716233de88bc: Already exists
27182ebf3982: Pull complete
c117476c1d40: Pull complete
937597159703: Pull complete
5da2fc5c703c: Pull complete
5a6b77fbd01e: Pull complete
Digest: sha256:5a4ecb2b8faafff4ff7e615308e791c5c32924e46bc0870256b5e0bb15952da7
Status: Downloaded newer image for chef/inspec:4.18.61
InSpec version: 4.18.61
Train version: 3.2.5
Command line configuration:
{"enable_telemetry"=>false,
 "winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "key_files"=>["secret_secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@10...",
 "chef_license"=>"accept-no-persist",
 "log_level"=>"debug",
 "diagnose"=>true}
JSON configuration file:
{"version"=>"1.2"}
Merged configuration:
{"reporter"=>["cli"],
 "show_progress"=>false,
 "color"=>true,
 "create_lockfile"=>true,
 "backend_cache"=>true,
 "enable_telemetry"=>false,
 "winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "key_files"=>["secret_secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@10...",
 "chef_license"=>"accept-no-persist",
 "log_level"=>"debug",
 "diagnose"=>true}

D, [2020-02-07T22:16:43.734052 #1] DEBUG -- : [SSH] opening connection to ec2-user@10...
D, [2020-02-07T22:16:43.734214 #1] DEBUG -- : [SSH] using options {:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}
D, [2020-02-07T22:16:44.116939 #1] DEBUG -- : [SSH] ec2-user@10... cmd = cmd.exe /c ver
D, [2020-02-07T22:16:44.163936 #1] DEBUG -- : [SSH] ec2-user@10... cmd = Get-WmiObject Win32_OperatingSystem | Select Caption,Version | ConvertTo-Json
D, [2020-02-07T22:16:44.209967 #1] DEBUG -- : [SSH] ec2-user@10... cmd = uname -s
D, [2020-02-07T22:16:44.258107 #1] DEBUG -- : [SSH] ec2-user@10... cmd = uname -m
D, [2020-02-07T22:16:44.303639 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/debian_version && cat /etc/debian_version
D, [2020-02-07T22:16:44.368166 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/os-release && cat /etc/os-release
D, [2020-02-07T22:16:44.426366 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /usr/bin/FastCli
D, [2020-02-07T22:16:44.472914 #1] DEBUG -- : [SSH] ec2-user@10... cmd = show version | json
D, [2020-02-07T22:16:44.516974 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/lsb-release && cat /etc/lsb-release
D, [2020-02-07T22:16:44.561485 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /usr/bin/lsb-release && cat /usr/bin/lsb-release
D, [2020-02-07T22:16:44.606237 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/oracle-release && cat /etc/oracle-release
D, [2020-02-07T22:16:44.650961 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/enterprise-release && cat /etc/enterprise-release
D, [2020-02-07T22:16:44.693962 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/parallels-release && cat /etc/parallels-release
D, [2020-02-07T22:16:44.740499 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/system-release && cat /etc/system-release
D, [2020-02-07T22:16:44.785723 #1] DEBUG -- : [SSH] ec2-user@10... cmd = bash -c '(sudo -v) < /dev/null'
^C/usr/local/bundle/gems/net-ssh-5.2.0/lib/net/ssh/connection/event_loop.rb:110:in `select': Interrupt
        from /usr/local/bundle/gems/net-ssh-5.2.0/lib/net/ssh/connection/event_loop.rb:110:in `ev_select_and_postprocess'
        from /usr/local/bundle/gems/net-ssh-5.2.0/lib/net/ssh/connection/event_loop.rb:31:in `process'
        from /usr/local/bundle/gems/net-ssh-5.2.0/lib/net/ssh/connection/session.rb:228:in `process'
        from /usr/local/bundle/gems/net-ssh-5.2.0/lib/net/ssh/connection/session.rb:181:in `block in loop'
        from /usr/local/bundle/gems/net-ssh-5.2.0/lib/net/ssh/connection/session.rb:181:in `loop'
        from /usr/local/bundle/gems/net-ssh-5.2.0/lib/net/ssh/connection/session.rb:181:in `loop'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/transports/ssh_connection.rb:329:in `execute_on_channel'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/transports/ssh_connection.rb:235:in `run_command_via_connection'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/plugins/base_connection.rb:128:in `run_command'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/extras/command_wrapper.rb:76:in `block in verify'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/extras/command_wrapper.rb:60:in `with_sudo_pty'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/extras/command_wrapper.rb:75:in `verify'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/extras/command_wrapper.rb:194:in `load'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/transports/ssh_connection.rb:57:in `initialize'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/transports/ssh.rb:240:in `new'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/transports/ssh.rb:240:in `create_new_connection'
        from /usr/local/bundle/gems/train-3.2.5/lib/train/transports/ssh.rb:82:in `connection'
        from /usr/local/bundle/gems/inspec-4.18.61/lib/inspec/backend.rb:53:in `create'
        from /usr/local/bundle/gems/inspec-4.18.61/lib/inspec/runner.rb:86:in `configure_transport'
        from /usr/local/bundle/gems/inspec-4.18.61/lib/inspec/runner.rb:78:in `initialize'
        from /usr/local/bundle/gems/inspec-4.18.61/lib/inspec/cli.rb:284:in `new'
        from /usr/local/bundle/gems/inspec-4.18.61/lib/inspec/cli.rb:284:in `exec'
        from /usr/local/bundle/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
        from /usr/local/bundle/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
        from /usr/local/bundle/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
        from /usr/local/bundle/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
        from /usr/local/bundle/gems/inspec-4.18.61/lib/inspec/base_cli.rb:35:in `start'
        from /usr/local/bundle/gems/inspec-bin-4.18.61/bin/inspec:11:in `<top (required)>'
        from /usr/local/bundle/bin/inspec:23:in `load'
        from /usr/local/bundle/bin/inspec:23:in `<main>'
```

And finally a run using 4.18:current (4.18.85) which results in a sudo failure:
```
docker run --rm -v $(pwd)/../inspec_files/:/share:Z chef/inspec:current exec -i secret_secret --sudo --target=ssh://ec2-user@10... --chef-license=accept-no-persist --no-enable-telemetry -l debug --diagnose az2-base-test.rb
Unable to find image 'chef/inspec:current' locally
current: Pulling from chef/inspec
Digest: sha256:5c29a641e496ca1b89a270aba0a4d351cc632d5f9d1738ab065dd52eba6ac989
Status: Downloaded newer image for chef/inspec:current
InSpec version: 4.18.85
Train version: 3.2.14
Command line configuration:
{"enable_telemetry"=>false,
 "winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "key_files"=>["secret_secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@10...",
 "chef_license"=>"accept-no-persist",
 "log_level"=>"debug",
 "diagnose"=>true}
JSON configuration file:
{"version"=>"1.2"}
Merged configuration:
{"reporter"=>["cli"],
 "show_progress"=>false,
 "color"=>true,
 "create_lockfile"=>true,
 "backend_cache"=>true,
 "enable_telemetry"=>false,
 "winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "key_files"=>["secret_secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@10...",
 "chef_license"=>"accept-no-persist",
 "log_level"=>"debug",
 "diagnose"=>true}

D, [2020-02-07T22:19:06.946042 #1] DEBUG -- : [SSH] opening connection to ec2-user@10...
D, [2020-02-07T22:19:06.946224 #1] DEBUG -- : [SSH] using options {:user_known_hosts_file=>"/dev/null", :port=>22, :compression=>false, :compression_level=>0, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :auth_methods=>["none", "publickey"], :keys_only=>true, :keys=>["secret_secret"], :password=>"<hidden>", :forward_agent=>nil, :non_interactive=>false, :verify_host_key=>:never}
D, [2020-02-07T22:19:07.259632 #1] DEBUG -- : [SSH] ec2-user@10... cmd = cmd.exe /c ver
D, [2020-02-07T22:19:07.306966 #1] DEBUG -- : [SSH] ec2-user@10... cmd = Get-WmiObject Win32_OperatingSystem | Select Caption,Version | ConvertTo-Json
D, [2020-02-07T22:19:07.354316 #1] DEBUG -- : [SSH] ec2-user@10... cmd = uname -s
D, [2020-02-07T22:19:07.401195 #1] DEBUG -- : [SSH] ec2-user@10... cmd = uname -m
D, [2020-02-07T22:19:07.447487 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/debian_version && cat /etc/debian_version
D, [2020-02-07T22:19:07.492426 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/os-release && cat /etc/os-release
D, [2020-02-07T22:19:07.538005 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /usr/bin/FastCli
D, [2020-02-07T22:19:07.582361 #1] DEBUG -- : [SSH] ec2-user@10... cmd = show version | json
D, [2020-02-07T22:19:07.628602 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/lsb-release && cat /etc/lsb-release
D, [2020-02-07T22:19:07.673620 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /usr/bin/lsb-release && cat /usr/bin/lsb-release
D, [2020-02-07T22:19:07.719627 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/oracle-release && cat /etc/oracle-release
D, [2020-02-07T22:19:07.764856 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/enterprise-release && cat /etc/enterprise-release
D, [2020-02-07T22:19:07.810455 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/parallels-release && cat /etc/parallels-release
D, [2020-02-07T22:19:07.859541 #1] DEBUG -- : [SSH] ec2-user@10... cmd = test -f /etc/system-release && cat /etc/system-release
D, [2020-02-07T22:19:07.906498 #1] DEBUG -- : [SSH] ec2-user@10... cmd = sh -c '(sudo -v) < /dev/null'
Sudo failed: We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

sudo: no tty present and no askpass program specified
```

There are changelog entries for 4.18.51, and 4.18.85, but I don't see any changes that would cause abberant behavior with ssh/sudo.

**Comment by n0coast** [comment]
This behavior is no longer present, using inspec/inspec:stable works as advertised.
Additionally chef_license and enable_telemetry seem to have been removed.

**Comment by james-stocks** [comment]
Are you sure @n0coast ? Is that a habitat package?
I think this still needs fixed - https://github.com/inspec/train/pull/576 has been suggested as a fix

**Comment by clintoncwolfe** [comment]
Our current understanding of the issue is that the latter breakage originated in a change in train gem 3.2.14 (which was then packaged in the docker image for inspec 4.18.65).

This isn't limited to Docker; any InSpec package that included train 3.2.14 or later is affected.

The current release of inspec, 4.1.8.97, includes train 4.2.22 which does not include any fix for the sudo problem. Also, to my knowledge, `--chef-license` and  `--enable-telemetry` were not removed - if so, that's a new bug.

**Comment by n0coast** [comment]
All I know is that using the current :stable docker image I was able to get my tests running that were previously broken but that I had to remove the `--chef-license` and `--enable-telemetry` flags - From what I can tell those two settings were removed based on debug output that no longer includes those options, previously and documented above in my command output you could see
```
 "enable_telemetry"=>false,
 "chef_license"=>"accept-no-persist",
```

Current output (without specifying those options) does not list them, and they are not documented in `--help` output:
```
docker run --rm -v inspec_files/:/share:Z chef/inspec:stable exec -l debug az2-base.rb -i secret --sudo -t ssh://ec2-user@... --diagnose
InSpec version: 3.9.3
Train version: 1.7.6
Command line configuration:
{"winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "log_level"=>"debug",
 "key_files"=>["secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@...",
 "diagnose"=>true}
JSON configuration file:
{"version"=>"1.1"}
Merged configuration:
{"reporter"=>["cli"],
 "show_progress"=>false,
 "color"=>true,
 "create_lockfile"=>true,
 "backend_cache"=>true,
 "winrm_transport"=>"negotiate",
 "insecure"=>false,
 "distinct_exit"=>true,
 "log_level"=>"debug",
 "key_files"=>["secret"],
 "sudo"=>true,
 "target"=>"ssh://ec2-user@.....",
 "diagnose"=>true}
```

This could be user error, if you have specific things I can do to help diagnose please let me know.

**Comment by clintoncwolfe** [comment]
Looks like you now have an older version of InSpec - 3.9.3.

**Comment by clintoncwolfe** [comment]
Circling back to the original issue, this is believed to be fixed in train 3.2.23; next release of inspec should be clear of the issue.

**Comment by n0coast** [comment]
Good catch on the inspec version - I assumed `stable` (3.9.3) would be closer to `current` (4.18.98) or `latest` (4.18.97), but it appears to be quite a bit behind, apologies for confusing the issue with my ignorance. 
I just tested with `chef/inspec:current` (4.18.98) and confirmed the sudo issue still exists and the license and telemetry options are valid.

**Comment by clintoncwolfe** [comment]
It is believed to be resolved in 4.18.100 .

**Comment by n0coast** [comment]
Working in chef/inspec:current (4.18.100).
